cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,170 @@
+    'edx-ui-toolkit/js/utils/html-utils',
+    'edx-ui-toolkit/js/utils/string-utils',
+    'js/views/utils/xblock_utils',
+    'text!templates/move-xblock-list.underscore'
+],
+function($, Backbone, _, gettext, HtmlUtils, StringUtils, XBlockUtils, MoveXBlockListViewTemplate) {
+    'use strict';
+
+    var XBlockListView = Backbone.View.extend({
+        el: '.xblock-list-container',
+
+        // parent info of currently displayed childs
+        parentInfo: {},
+        // child info of currently displayed child XBlocks
+        childrenInfo: {},
+        // list of visited parent XBlocks, needed for backward navigation
+        visitedAncestors: null,
+
+        // parent to child relation map
+        categoryRelationMap: {
+            course: 'section',
+            section: 'subsection',
+            subsection: 'unit',
+            unit: 'component'
+        },
+
+        categoriesText: {
+            section: gettext('Sections'),
+            subsection: gettext('Subsections'),
+            unit: gettext('Units'),
+            component: gettext('Components')
+        },
+
+        events: {
+            'click .button-forward': 'renderChildren'
+        },
+
+        initialize: function(options) {
+            this.visitedAncestors = [];
+            this.template = HtmlUtils.template(MoveXBlockListViewTemplate);
+            this.ancestorInfo = options.ancestorInfo;
+            this.listenTo(Backbone, 'move:breadcrumbButtonPressed', this.handleBreadcrumbButtonPress);
+            this.renderXBlockInfo();
+        },
+
+        render: function() {
+            HtmlUtils.setHtml(
+                this.$el,
+                this.template(
+                    {
+                        xblocks: this.childrenInfo.children,
+                        noChildText: this.getNoChildText(),
+                        categoryText: this.getCategoryText(),
+                        XBlocksCategory: this.childrenInfo.category,
+                        forwardButtonSRText: this.getForwardButtonSRText(),
+                        currentLocationIndex: this.getCurrentLocationIndex()
+                    }
+                )
+            );
+            Backbone.trigger('move:childrenRendered', this.breadcrumbInfo());
+            return this;
+        },
+
+        renderChildren: function(event) {
+            this.renderXBlockInfo(
+                'forward',
+                $(event.target).closest('.xblock-item').data('itemIndex')
+            );
+        },
+
+        handleBreadcrumbButtonPress: function(newParentIndex) {
+            this.renderXBlockInfo('backward', newParentIndex);
+        },
+
+        renderXBlockInfo: function(direction, newParentIndex) {
+            if (direction === undefined) {
+                this.parentInfo.parent = this.model;
+            } else if (direction === 'forward') {
+                // clicked child is the new parent
+                this.parentInfo.parent = this.childrenInfo.children[newParentIndex];
```

Comment:
this.parentInfo is an object which contains parent XBlock info plus parent category